harp/reader.py

```diff
@@ -44,13 +44,53 @@ def __init__(
         self.read = read
 
 
+class RegisterMap:
```

Comment:
Consider subclassing from `dict`, `UserDict` or `MutableMapping` to make it clear that we are defining a collection object, and to have objects that can always be used just like dicts (ensure we did not forget to define anything).

See [this SO answer](https://stackoverflow.com/a/7148602/628228) for some interesting considerations.

Reply:
Went with UserDict, seems like the correct pick in this case as we overload the index operator.